packages/fxa-auth-server/test/local/payments/configuration/manager.js

```diff
@@ -0,0 +1,260 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const { assert } = require('chai');
+const { default: Container } = require('typedi');
+const { cloneDeep } = require('lodash');
+const retry = require('async-retry');
+
+const {
+  AuthFirestore,
+  AuthLogger,
+  AppConfig,
+} = require('../../../../lib/types');
+const {
+  PaymentConfigManager,
+} = require('../../../../lib/payments/configuration/manager');
+const { setupFirestore } = require('../../../../lib/firestore-db');
+const { randomUUID } = require('crypto');
+const errors = require('../../../../lib/error');
+
+const mockConfig = {
+  authFirestore: {
+    prefix: 'mock-fxa-',
+  },
+  subscriptions: {
+    playApiServiceAccount: {
+      credentials: {
+        clientEmail: 'mock-client-email',
+      },
+      keyFile: 'mock-private-keyfile',
+    },
+  },
+};
+
+const productConfig = {
+  active: true,
+  stripeProductId: 'mock-stripe-product-id',
+  capabilities: {
+    '*': 'stuff',
+  },
+  locales: {},
+  styles: {
+    webIconBackground: '#fff',
+  },
+  support: {},
+  uiContent: {},
+  urls: {
+    download: 'https://download.com',
+    privacyNotice: 'https://privacy.com',
+    termsOfService: 'https://terms.com',
+    termsOfServiceDownload: 'https://terms-download.com',
+    webIcon: 'https://web-icon.com',
+  },
+};
+
+const planConfig = {
+  active: true,
+  stripePriceId: 'mock-stripe-price-id',
+  capabilities: {
+    '*': 'stuff',
+  },
+  locales: {},
+  styles: {
+    webIconBackground: '#fff',
+  },
+  support: {},
+  uiContent: {},
+};
+
+async function deleteQueryBatch(db, query, resolve) {
+  const snapshot = await query.get();
+
+  const batchSize = snapshot.size;
+  if (batchSize === 0) {
+    // When there are no documents left, we are done
+    resolve();
+    return;
+  }
+
+  // Delete documents in a batch
+  const batch = db.batch();
+  snapshot.docs.forEach((doc) => {
+    batch.delete(doc.ref);
+  });
+  await batch.commit();
+
+  // Recurse on the next process tick, to avoid
+  // exploding the stack.
+  process.nextTick(() => {
+    deleteQueryBatch(db, query, resolve);
+  });
```

Comment:
Can you clarify what this is doing? Per the docs, this ensures the callback is executed before the next tick of the event loop. Without this here, all of these recursive calls would occur in the same tick of the event loop?

Reply:
It's a recursive call, if there were a ton of calls, you can hit recursion limits. Pushing off the function to run in the next tick breaks the stack, so it could iterate as long as needed. This code is copied from the docs here:
https://firebase.google.com/docs/firestore/manage-data/delete-data#collections